sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -281,33 +361,25 @@ case class DataSource(
       // This is a non-streaming file based datasource.
       case (format: FileFormat, _) =>
         val allPaths = caseInsensitiveOptions.get("path") ++ paths
+        val hadoopConf = sparkSession.sessionState.newHadoopConf()
         val globbedPaths = allPaths.flatMap { path =>
```

Comment:
Just to confirm, for paths with globs, this would expand here ONCE, and then expand them AGAIN in `getOrInferFileFormatSchema`, right? 

If so, we dont have to fix it in this PR, but we should document this in a JIRA or something for fixing later.